static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,87 @@
+import {createContext, useCallback, useContext, useEffect, useState} from 'react';
+
+import {
+  DrawerBody,
+  DrawerContainer,
+  DrawerPanel,
+} from 'sentry/components/globalDrawer/components';
+import type {
+  DrawerConfig,
+  DrawerContext as DrawerUsageContext,
+  DrawerContextProps,
+} from 'sentry/components/globalDrawer/types';
+import {useLocation} from 'sentry/utils/useLocation';
+
+const DEFAULT_DRAWER_CONTEXT: DrawerContextProps = {
+  config: {
+    renderer: null,
+    options: {
+      closeOnEscapeKeypress: true,
+      closeOnOutsideClick: true,
+    },
+  },
+  openDrawer: () => {},
+  closeDrawer: () => {},
+};
+
+const DrawerContext = createContext<DrawerContextProps>(DEFAULT_DRAWER_CONTEXT);
+
+export function DrawerContextProvider({children}) {
+  const location = useLocation();
+  const [drawerConfig, setDrawerConfig] = useState<DrawerConfig>(
+    DEFAULT_DRAWER_CONTEXT.config
+  );
+  const openDrawer = useCallback(
+    (renderer, options = {}) => setDrawerConfig({renderer, options}),
+    [setDrawerConfig]
```

Comment:
I actually think you don't even need to put this in here, the eslint checker knows that `setDrawerConfig` will never change reference I think
```suggestion
    []
```